### PR TITLE
Fix turn allocation latency

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Ice/IceUtils.c [hunk 1]
```diff
@@ -224,6 +224,18 @@ STATUS populateIpFromString(PKvsIpAddress pKvsIpAddress, PCHAR pBuff)
     return retStatus;
 }
 
+VOID iceUtilGenerateRandomStr(PCHAR buf, UINT32 bufLen)
```
Comment (Sean-Der) [inline src/source/Ice/IceUtils.c:227]: Can you use the random string generation we already have?

File: src/source/Ice/SocketConnection.c [hunk 2]
```diff
@@ -109,8 +110,6 @@ STATUS socketConnectionInitSecureConnection(PSocketConnection pSocketConnection,
     SSL_CTX_set_read_ahead(pSocketConnection->pSslCtx, 1);
     SSL_CTX_set_verify(pSocketConnection->pSslCtx, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, certificateVerifyCallback);
 
-    CHK(SSL_CTX_set_cipher_list(pSocketConnection->pSslCtx, "HIGH:!aNULL:!MD5:!RC4"), STATUS_SSL_CTX_CREATION_FAILED);
```
Comment (Sean-Der) [inline src/source/Ice/SocketConnection.c:112]: Why remove?

**Comment by Sean-Der** [comment]
@chehefen Can you break out your additional logging and performance changes into unique commits? It is much harder to annotate/debug otherwise.

really nice work!

File: src/source/Ice/SocketConnection.c (hunk 2 above)
Comment (chehefen) [reply src/source/Ice/SocketConnection.c:112]: oh sry, will add back.

**Comment by chehefen** [comment]
replaced with #386

**Pull request closed without merging**